Code/Python/linked_list_intersection.py

```diff
@@ -0,0 +1,136 @@
+#Problem-Statement: Given two heads of singly linked-lists headA and headB, 
+#return the node at which the two linked lists intersect.
+
+#	 A :    3->7->2
+#	               \
+#		        5->6->4
+#		       /
+# 	 B : 1->9->0->8
+
+# The node intersect at node with value 5. 
+# Note: Do not alter the original structure of linked list after function returns.
+
+# Node class 
+class Node:  
+    # Function to initialise the node object 
+    def __init__(self, data): 
+        self.val = data  # Assign data 
+        self.next = None  # Initialize next as null 
+  
+  
+# Linked List class contains a Node object 
+class LinkedList:  
+    # Function to initialize head 
+    def __init__(self): 
+        self.head = None
+        
+
+# 1. Brute force approach: 
+# For each node in linkedlist A, traverse over other linkedlist B, 
+# and check whether the node is present in B.
+# Time complexity: O(m*n) ~ O(n^2), Space complexity: O(1)
+# Caution: We are comparing nodes here, not value present in each node. Value can repeat.
+
+def getIntersectNode(headA, headB):
+	ptr1 = headA
+	while ptr1:
+		ptr2 = headB
+		while ptr2:
+			if ptr1 == ptr2:
+				return ptr1
+			ptr2 = ptr2.next
+		ptr1 = ptr1.next
+	return None # if two linked list do not intersect
+	
+# 2. Using two pointers
+# The two linkedlist might not have same length,
+# Therefore, we must jump few steps ahead in longer list, so that length with smaller linkedlist 
+# becomes equal and start comparing simultaneously to reach a common at common Node.
+# Algorithm : With 4 loops, one after the other, each doing the following:
+# the length of list A.
+# the length of list B.
+# Set the start pointer for the longer list.
+# Step the pointers through the list together.
+
+def twoPointerIntersect(headA, headB):
+    ptr1 = headA
+    ptr2 = headB
+    n1=n2=0
+    while ptr1:
+        n1 +=1
+        ptr1 = ptr1.next
+    
+    while ptr2:
+        n2 += 1
+        ptr2 = ptr2.next
+    
+    diff = abs(n2-n1)
+    c = 0
+    ptr1 = headA
+    ptr2 = headB
+    if n2 < n1:
+        while ptr1:
+            if c==diff:
+                break
+            ptr1 = ptr1.next
+            c+=1
+        while ptr1 and ptr2:
+            if ptr1==ptr2:
+                return ptr1
+            ptr1 = ptr1.next
+            ptr2 = ptr2.next
+        return None           
+        
+    else:
+        while ptr2:
+            if c==diff:
+                break
+            ptr2 = ptr2.next
+            c+=1
+        while ptr1 and ptr2:
+            if ptr1==ptr2:
+                return ptr1
+            ptr1 = ptr1.next
+            ptr2 = ptr2.next
+        return None
+ 
+
+if __name__ == '__main__':
+	listA = LinkedList()
+	listA.head = Node(3)
+	node2a = Node(7)
+	node3a = Node(2)
```

Comment:
make your program to take dynamic inputs

Reply:
Hello, updated the code for dynamic inputs.